c/driver/common/utils.c

```diff
@@ -884,7 +884,7 @@ struct AdbcGetObjectsConstraint* AdbcGetObjectsDataGetConstraintByName(
     for (int64_t i = 0; i < table->n_table_constraints; i++) {
       struct AdbcGetObjectsConstraint* constraint = table->table_constraints[i];
       struct ArrowStringView name = constraint->constraint_name;
-      if (!strncmp(name.data, constraint_name, name.size_bytes)) {
+      if (!constraint_name || !strncmp(name.data, constraint_name, name.size_bytes)) {
```

Comment:
Hmm is this function receiving a nullptr? Seems a bit strange to call a `GetByName` function, pass it a NULL pointer for the name and expect a result back

Reply:
Current implementation does allow for UB if `constraint_name` is NULL so makes sense to return early if that happens, though I think that should be on function entry and also just return NULL there instead of grabbing the first constraint?

Reply:
Fair. 

SQLite doesn't name its constraints so I wanted some way to just get any constraint.

Reply:
Ah OK. In that case maybe you want to do something like this instead?

```c
  struct AdbcGetObjectsTable* table =
      AdbcGetObjectsDataGetTableByName(*get_objects_data, quirks()->catalog().c_str(),
                                       quirks()->db_schema().c_str(), "adbc_pkey_test");
  if (table->n_table_constraints > 0) {
    struct AdbcGetObjectsConstraint *constraint = table->table_constraints[0];
    ...
  }
```

Reply:
Alternately could add a `AdbcGetObjectsDataGetConstraintByPosition` function

Reply:
Ah, nice. I'll update it tomorrow morning